File: challenges/3-tipos-de-triangulo/typescript/marcosvag/tiposDeTriangulo.ts
const lados = process.argv.slice(2).map((elemento:string) => parseFloat(elemento));

lados.sort((a:number, b:number) => b - a);

function acharTipoDoTriangulo([ladoA, ladoB, ladoC]:number[]):void {
    if(ladoA >= ladoB + ladoC) {
        console.log('NAO FORMA TRIANGULO')
        return;
    }
    if(ladoA ** 2 == (ladoB ** 2 + ladoC ** 2)) {
        console.log('TRIANGULO RETANGULO')
    }
    if(ladoA ** 2 > (ladoB ** 2 + ladoC ** 2)) {
        console.log('TRIANGULO OBTUSANGULO');
    }
    if(ladoA ** 2 < (ladoB ** 2 + ladoC ** 2)) {
        console.log('TRIANGULO ACUTANGULO');
    }
    if(ladoA == ladoB && ladoB == ladoC) {
        console.log('TRIANGULO EQUILATERO');   
    }
    if(ladoA == ladoB && ladoA != ladoC || ladoB == ladoC && ladoB != ladoA || ladoC == ladoA && ladoC != ladoB){
        console.log('TRIANGULO ISOSCELES');
    }
    return;
}
acharTipoDoTriangulo(lados)

File: challenges/3-tipos-de-triangulo/java/camilealheiro/TiposDeTriangulo.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Locale;
import java.util.Scanner;

import static java.lang.Math.pow;

public class TiposDeTriangulo {
    public static void main(String[] args) {

        ArrayList<Float> lados = new ArrayList<>();
        Scanner sc = new Scanner(System.in);
        sc.useLocale(Locale.US);
        System.out.println("Digite três números separados por espaço: ");

        lados.add(sc.nextFloat());
        lados.add(sc.nextFloat());
        lados.add(sc.nextFloat());

        lados.sort(Collections.reverseOrder());

        float A = lados.get(0);
        float B = lados.get(1);
        float C = lados.get(2);

        double sumQuadrados = pow(B, 2) + pow(C, 2);

        if(A >= B + C)
            System.out.println("NAO FORMA TRIANGULO");
        else {
            if(pow(A, 2) == sumQuadrados)
                System.out.println("TRIANGULO RETANGULO");
            if(pow(A, 2) > sumQuadrados)
                System.out.println("TRIANGULO OBTUSANGULO");
            if(pow(A, 2) < sumQuadrados)
                System.out.println("TRIANGULO ACUTANGULO");
            if(A == B && A == C)
                System.out.println("TRIANGULO EQUILATERO");
            if((A == B || A == C || B == C) && (A != B || A != C || B != C))
                System.out.println("TRIANGULO ISOSCELES");
        }
    }
}
